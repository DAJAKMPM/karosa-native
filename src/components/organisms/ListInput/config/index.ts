/**
 *
 * @format
 *
 */

export const variationOneLines = 1;
export const variationTwoLines = 10;
