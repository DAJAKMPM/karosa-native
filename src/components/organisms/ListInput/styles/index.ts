/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";
import { DIMENS, SPACING, theme } from "@app/styles";

const ListInputStyles = StyleSheet.create({
  variationOneContainer: {
    flexDirection: "column",
  },
  variationTwoContainer: {
    flexDirection: "column",
  },
  formInputContainer: {
    width: "100%",
    marginLeft: "3%",
  },
  mainRowContainer: {
    flexDirection: "row",
    height: DIMENS.screenHeight * 0.028,
  },
  labelLengthContainer: {
    flexDirection: "row",
  },
  labelContainer: {
    flexDirection: "row",
    justifyContent: "flex-start",
  },
  icon: {
    position: "relative",
    left: 0,
    top: -1,
  },
  infoContainer: {
    width: "90%",
    height: DIMENS.screenHeight * 0.07,
    flexDirection: "row",
  },
  rowInputContainer: {
    position: "relative",
  },
  inputContainer: {
    borderWidth: 0,
    borderBottomWidth: 0,
    marginTop: 0,
    marginBottom: 0,
    height: 0,
  },
  columnInputContainer: {
    borderWidth: 0,
    borderBottomWidth: 0,
    padding: 0,
    marginBottom: 6,
  },
  columnContainer: {
    position: "relative",
    bottom: -12,
    left: -11,
  },
  txtLabel: {
    ...theme.textListItem,
    color: theme.colors.dark20,
    fontWeight: "400",
  },
  txtValue: {
    ...theme.textListItem,
    color: theme.colors.primary,
    fontWeight: "400",
  },
  txtInfo: {
    ...theme.textRegular,
    color: theme.colors.dark10,
    fontWeight: "400",
  },
  txtLength: {
    ...theme.textListItem,
    fontWeight: "400",
    color: theme.colors.dark10,
    marginLeft: 2,
  },
  txtRequired: {
    ...theme.textRegular,
    marginLeft: SPACING.xs,
    color: theme.colors.red5,
    fontWeight: "400",
  },
  txtWithIcon: {
    ...theme.textListItem,
    position: "relative",
    left: 7,
    marginRight: 6,
    fontWeight: "400",
  },
  txtRowInput: {
    ...theme.textListItem,
    textAlign: "right",
    color: theme.colors.primary,
    fontWeight: "400",
  },
});

export default ListInputStyles;
