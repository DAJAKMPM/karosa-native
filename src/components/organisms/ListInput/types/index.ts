/**
 *
 * @format
 *
 */

import { KeyboardTypeOptions } from "react-native";

export type PropsType = {
  variation: number;
  label: string;
  name: string;
  maxLen?: number;
  placeholder: string;
  required?: boolean;
  hasBottomDivider?: boolean;
  keyboardType?: KeyboardTypeOptions;
};
