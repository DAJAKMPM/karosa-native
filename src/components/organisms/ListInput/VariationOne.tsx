/**
 *
 * VariationOne
 * @format
 *
 */

import React, { FC } from "react";
import { useField } from "formik";
import { ListItem } from "react-native-elements";
import Text from "@app/atoms/Text";
import FormInput from "@app/molecules/FormInput";
import ValidationMessage from "@app/components/molecules/ValidationMessage";

import type { PropsType } from "./types";
import { NUM_LINES } from "./config";
import ListInputStyles from "./styles";

const VariationOne: FC<PropsType> = (props) => {
  const {
    label,
    hasBottomDivider,
    name,
    placeholder,
    required,
    maxLen,
    keyboardType,
  } = props;

  const [, meta] = useField(name);
  const currentLength = String(meta.value).length;

  return (
    <ListItem bottomDivider={hasBottomDivider}>
      <ListItem.Content style={ListInputStyles.variationOneContainer}>
        <ListItem.Content style={ListInputStyles.labelLengthContainer}>
          <Text text={label} customStyle={ListInputStyles.txtLabel} />
          <Text
            text={`${currentLength}/${maxLen}`}
            customStyle={ListInputStyles.txtLength}
          />
          {required && (
            <Text text="*" customStyle={ListInputStyles.txtRequired} />
          )}
        </ListItem.Content>
        <FormInput
          name={name}
          keyboardType={keyboardType}
          placeholder={placeholder}
          numberOfLines={NUM_LINES.TEN}
          multiline
          inputContainerStyle={ListInputStyles.columnInputContainer}
        />
        {meta.error && (
          <ListItem.Content style={ListInputStyles.errorContainer}>
            <ValidationMessage name={name} />
          </ListItem.Content>
        )}
      </ListItem.Content>
    </ListItem>
  );
};

export default VariationOne;
