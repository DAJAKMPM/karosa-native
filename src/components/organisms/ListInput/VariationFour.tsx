import FormInput from "@app/components/molecules/FormInput";
import React, { FC } from "react";
import { View } from "react-native";
import { Text } from "react-native-elements";
import { useField } from "formik";
import ListInputStyles from "./styles";

import type { PropsType } from "./types";
import { theme } from "@app/styles";

const VariationFour: FC<PropsType> = (props) => {
  const { label, name, placeholder, required, maxLen, keyboardType } = props;

  const [, meta] = useField(name);

  const currentLength = String(meta.value).length;

  return (
    <View>
      <View style={{ flexDirection: "row" }}>
        <Text style={{ marginRight: 2, fontSize: 13.33, lineHeight: 16 }}>
          {label}
        </Text>
        <Text
          style={{
            ...theme.textListItem,
            fontWeight: "400",
            color: theme.colors.dark10,
            marginLeft: 2,
          }}>{`(${currentLength}/${maxLen})`}</Text>
        {required && <Text style={{ color: theme.colors.red5 }}>*</Text>}
      </View>

      <FormInput
        name={name}
        keyboardType={keyboardType}
        placeholder={placeholder}
        numberOfLines={1}
        multiline={false}
        inputStyle={ListInputStyles.txtValue}
        inputContainerStyle={ListInputStyles.columnInputContainer}
      />
    </View>
  );
};

export default VariationFour;
