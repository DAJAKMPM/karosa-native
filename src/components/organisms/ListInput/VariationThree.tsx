/**
 *
 * VariationThree
 * @format
 *
 */

import React, { FC } from "react";
import { ListItem } from "react-native-elements";
import { View } from "react-native";
import { useFieldError } from "@app/hooks";
import { theme } from "@app/styles";
import Text from "@app/atoms/Text";
import FormInput from "@app/molecules/FormInput";
import ValidationMessage from "@app/molecules/ValidationMessage";

import type { PropsType } from "./types";
import { NUM_LINES } from "./config";
import ListInputStyles from "./styles";

const VariationThree: FC<PropsType> = (props) => {
  const {
    label,
    name,
    placeholder,
    required,
    hasBottomDivider,
    keyboardType,
    info,
  } = props;

  const { isError } = useFieldError(name);

  return (
    <ListItem bottomDivider={hasBottomDivider}>
      <ListItem.Content style={ListInputStyles.variationTwoContainer}>
        <View style={ListInputStyles.mainRowContainer}>
          <View style={ListInputStyles.labelContainer}>
            <Text text={label} textStyle={ListInputStyles.txtLabel} />
            {required && (
              <Text text="*" textStyle={ListInputStyles.txtRequired} />
            )}
          </View>
        </View>
        <View style={ListInputStyles.infoContainer}>
          {info && <Text text={info} textStyle={ListInputStyles.txtInfo} />}
        </View>
        <View style={ListInputStyles.mainRowContainer}>
          <ListItem.Content style={ListInputStyles.rowInputContainer}>
            <FormInput
              keyboardType={keyboardType}
              name={name}
              placeholder={placeholder}
              placeholderColor={theme.colors.primary}
              numberOfLines={NUM_LINES.ONE}
              inputStyle={ListInputStyles.txtRowInput}
              inputContainerStyle={ListInputStyles.inputContainer}
            />
          </ListItem.Content>
        </View>

        {isError && <ValidationMessage name={name} />}
      </ListItem.Content>
    </ListItem>
  );
};

export default VariationThree;
