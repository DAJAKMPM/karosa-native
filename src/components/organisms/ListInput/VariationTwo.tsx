/**
 *
 * VariationTwo
 * @format
 *
 */

import React, { FC } from "react";
import { ListItem } from "react-native-elements";
import { View } from "react-native";
import { theme } from "@app/styles";
import Text from "@app/atoms/Text";
import FormInput from "@app/molecules/FormInput";
import ValidationMessage from "@app/molecules/ValidationMessage";

import type { PropsType } from "./types";
import { NUM_LINES } from "./config";
import ListInputStyles from "./styles";

const VariationTwo: FC<PropsType> = (props) => {
  const { label, name, placeholder, required, hasBottomDivider, keyboardType } =
    props;

  return (
    <ListItem bottomDivider={hasBottomDivider}>
      <ListItem.Content style={ListInputStyles.variationTwoContainer}>
        <ListItem.Content style={ListInputStyles.mainRowContainer}>
          <View style={ListInputStyles.labelContainer}>
            <Text text={label} textStyle={ListInputStyles.txtLabel} />
            {required && (
              <Text text="*" textStyle={ListInputStyles.txtRequired} />
            )}
          </View>
          <ListItem.Content style={ListInputStyles.rowInputContainer}>
            <FormInput
              keyboardType={keyboardType}
              name={name}
              placeholder={placeholder}
              placeholderColor={theme.colors.primary}
              numberOfLines={NUM_LINES.ONE}
              inputStyle={ListInputStyles.txtRowInput}
              inputContainerStyle={ListInputStyles.inputContainer}
            />
          </ListItem.Content>
        </ListItem.Content>
        <ValidationMessage name={name} />
      </ListItem.Content>
    </ListItem>
  );
};

export default VariationTwo;
