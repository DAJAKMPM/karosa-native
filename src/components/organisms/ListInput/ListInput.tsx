/**
 *
 * VariationOne
 * @format
 *
 */

import React, { FC } from "react";

import type { PropsType } from "./types";
import VariationOne from "./VariationOne";
import VariationTwo from "./VariationTwo";

const ListInput: FC<PropsType> = (props) => {
  const {
    variation,
    hasBottomDivider,
    label,
    name,
    placeholder,
    required,
    maxLen,
    keyboardType,
  } = props;

  switch (variation) {
    case 1:
      return (
        <VariationOne
          variation={variation}
          label={label}
          hasBottomDivider={hasBottomDivider}
          name={name}
          placeholder={placeholder}
          required={required}
          maxLen={maxLen}
          keyboardType={keyboardType}
        />
      );
    case 2:
      return (
        <VariationTwo
          variation={variation}
          label={label}
          hasBottomDivider={hasBottomDivider}
          name={name}
          placeholder={placeholder}
          required={required}
          keyboardType={keyboardType}
        />
      );

    default:
      return (
        <VariationOne
          variation={variation}
          label={label}
          hasBottomDivider={hasBottomDivider}
          name={name}
          placeholder={placeholder}
          required={required}
          maxLen={maxLen}
          keyboardType={keyboardType}
        />
      );
  }
};

export default ListInput;
