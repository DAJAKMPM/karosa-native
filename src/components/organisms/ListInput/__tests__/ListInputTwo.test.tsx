import React from "react";
import { renderWithFormikWrapper } from "@app/ci/testHelper";
import ListInput from "../ListInput";
import { COMMON } from "@app/constants";

describe("ListInput variation two Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithFormikWrapper(
      { test: "" },
      <ListInput
        label="testLabel"
        name="test"
        placeholder="test"
        variation={COMMON.VARIATION.TWO}
        hasBottomDivider
        required
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
