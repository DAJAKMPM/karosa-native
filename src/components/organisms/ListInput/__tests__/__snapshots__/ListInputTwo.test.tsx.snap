// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListInput variation two Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderBottomWidth": 0.5,
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="padView"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "justifyContent": "center",
            },
            Object {
              "flexDirection": "column",
            },
          ]
        }
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "justifyContent": "center",
              },
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
            ]
          }
          theme={
            Object {
              "colors": Object {
                "black": "#242424",
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "grey": "rgba(0, 0, 0, 0.54)",
                    "primary": "#2196f3",
                    "searchBg": "#dcdce1",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "default": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "web": Object {
                    "error": "#ff190c",
                    "grey": "#393e42",
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
                "white": "#ffffff",
              },
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "flex": 1,
                  "justifyContent": "center",
                },
                Object {
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                },
              ]
            }
            theme={
              Object {
                "colors": Object {
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": Object {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1C1C1C",
                  "fontFamily": "SFProText-Regular",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "lineHeight": 16,
                }
              }
            >
              testLabel
            </Text>
            <Text
              style={
                Object {
                  "color": "#EB4837",
                  "fontFamily": "SFProText-Regular",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "lineHeight": 18.75,
                  "marginLeft": 4,
                }
              }
            >
              *
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "flex": 1,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
            theme={
              Object {
                "colors": Object {
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": Object {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
              }
            }
          >
            <View
              style={
                Object {
                  "paddingHorizontal": 0,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "borderBottomWidth": 0,
                    "borderColor": "#4F4F4F",
                    "borderRadius": 3,
                    "borderStyle": "solid",
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "height": 0,
                    "padding": 0,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  editable={true}
                  keyboardType="default"
                  numberOfLines={1}
                  onChangeText={[Function]}
                  placeholder="test"
                  placeholderTextColor="#0AA351"
                  rejectResponderTermination={true}
                  returnKeyType="done"
                  style={
                    Object {
                      "color": "#1C1C1C",
                      "flex": 1,
                      "fontSize": 14,
                      "fontWeight": "400",
                      "lineHeight": 16,
                      "minHeight": 40,
                      "textAlign": "right",
                    }
                  }
                  testID="RNE__Input__text-input"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <Text
                style={
                  Object {
                    "color": "#ff190c",
                    "display": "none",
                    "fontSize": 12,
                    "margin": 5,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
